src/gc-debug.c

```diff
@@ -651,39 +677,24 @@ void objprofile_count(void *ty, int old, int sz)
 
 void objprofile_reset(void)
 {
+    grab_objprofile = 0;
     for (int g = 0; g < 3; g++) {
         htable_reset(&obj_counts[g], 0);
         htable_reset(&obj_sizes[g], 0);
     }
 }
 
-static void objprofile_print(htable_t nums, htable_t sizes)
+static void objprofile_print(
+        JL_STREAM *out, char *str_timestamp, char *kind, htable_t nums, htable_t sizes)
 {
     for(int i=0; i < nums.size; i+=2) {
         if (nums.table[i+1] != HT_NOTFOUND) {
             void *ty = nums.table[i];
             int num = (intptr_t)nums.table[i + 1] - 1;
             size_t sz = (uintptr_t)ptrhash_get(&sizes, ty) - 1;
             static const int ptr_hex_width = 2 * sizeof(void*);
-            if (sz > 2e9) {
-                jl_safe_printf(" %6d : %*.1f GB of (%*p) ",
-                               num, 6, ((double)sz) / 1024 / 1024 / 1024,
-                               ptr_hex_width, ty);
-            }
-            else if (sz > 2e6) {
-                jl_safe_printf(" %6d : %*.1f MB of (%*p) ",
-                               num, 6, ((double)sz) / 1024 / 1024,
-                               ptr_hex_width, ty);
-            }
-            else if (sz > 2e3) {
-                jl_safe_printf(" %6d : %*.1f kB of (%*p) ",
-                               num, 6, ((double)sz) / 1024,
-                               ptr_hex_width, ty);
-            }
-            else {
-                jl_safe_printf(" %6d : %*d  B of (%*p) ",
-                          num, 6, (int)sz, ptr_hex_width, ty);
-            }
+            // timestamp,kind,num,size,{ptr_hex_with,ty},$objtype
+            jl_safe_printf( "OBJPROFILE;%s;%s;%s;%d;%ld;", objprofile_name, str_timestamp, kind, num, sz);
```

Comment:
Any reason not to use commas instead of `;`?

Reply:
Using commas makes the file a bit confusing for a downstream CSV parser, because the last column (object type) often has commas in it. So, we decided to use a different delimiter.

I suppose we could also solve this by surrounding the last column in quotes, so the commas are escaped.